management/database/client.go

```diff
@@ -0,0 +1,154 @@
+package database
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+)
+
+// Definitions of numerous constants representing API endpoints.
+const (
+	azureCreateDatabaseServerUrl = "services/sqlservers/servers"
+	azureListDatabaseServersUrl  = "/services/sqlservers/servers"
+	azureDeleteDatabaseServerUrl = "/services/sqlservers/servers/%s"
+
+	azureCreateDatabaseUrl = "services/sqlservers/servers/%s/databases"
+	azureGetDatabaseUrl    = "/services/sqlservers/servers/%s/databases/%s"
+	azureListDatabasesUrl  = "/services/sqlservers/servers/%s/databases?contentview=generic"
+	azureDeleteDatabaseUrl = "/services/sqlservers/servers/%s/databases/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// DatabaseClient defines various database CRUD operations.
+// It contains a management.Client for making the actual http calls.
+type DatabaseClient struct {
+	mgmtClient management.Client
+}
+
+// NewClient returns a new DatabaseClient struct with the provided
+// management.Client as the underlying client.
+func NewClient(mgmtClient management.Client) DatabaseClient {
+	return DatabaseClient{mgmtClient}
+}
+
+// CreateDatabaseServer issues a database server creation request to Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505699.aspx
+func (c DatabaseClient) CreateDatabaseServer(params DatabaseServerCreateParams) error {
+	if params.Location == "" {
+		return fmt.Errorf(errParamNotSpecified, "Location")
+	}
+	if params.UserName == "" {
+		return fmt.Errorf(errParamNotSpecified, "UserName")
+	}
+	if params.Password == "" {
+		return fmt.Errorf(errParamNotSpecified, "Password")
+	}
+	if params.Version == "" {
+		return fmt.Errorf(errParamNotSpecified, "")
+	}
+
+	req, err := xml.Marshal(params)
+	if err != nil {
+		return err
+	}
+
+	// NOTE: would be nice to be able to read the response here:
+	_, err = c.mgmtClient.SendAzurePostRequest(azureCreateDatabaseServerUrl, req)
+	return err
+}
+
+// ListDatabaseServers issues a request for a list of all database servers off Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505702.aspx
+func (c DatabaseClient) ListDatabaseServers() ([]DatabaseServer, error) {
+	var resp ListDatabaseServersResponse
+
+	data, err := c.mgmtClient.SendAzureGetRequest(azureListDatabaseServersUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(data, &resp)
+	return resp.Servers, err
```

Comment:
Do not return the internal field, just return the response object `ListDatabaseServersResponse` and let the user figure out.

If some other field gets added right next to `.Servers` field, we'll be introducing breaking change to return that as well, we follow returning xxxResponse objects as a convention in the codebase.